### PR TITLE
disttask: persist subtask error properly

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/46258

Problem Summary:

Previously, `UpdateErrorToSubtask` updates all subtasks regardless of global task ID.

### What is changed and how it works?

Add `taskID` parameter to `UpdateErrorToSubtask`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: disttask/framework/scheduler/interface.go [hunk 1]
```diff
@@ -26,12 +26,12 @@ type TaskTable interface {
 	GetGlobalTaskByID(taskID int64) (task *proto.Task, err error)
 
 	GetSubtaskInStates(instanceID string, taskID int64, step int64, states ...interface{}) (*proto.Subtask, error)
-	StartSubtask(id int64) error
-	UpdateSubtaskStateAndError(id int64, state string, err error) error
-	FinishSubtask(id int64, meta []byte) error
```
Comment (xhebox) [inline disttask/framework/scheduler/interface.go:31]: Do we also need to add for these three?

Comment (tangenta) [reply disttask/framework/scheduler/interface.go:31]: `subtaskID` is unique enough to locate the record.

**Review by xhebox — APPROVED** [review]

**Review by D3Hunter — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**